medmodels/treatment_effect/matching/tests/test_matching.py

```diff
@@ -0,0 +1,241 @@
+        self.assertEqual(len(matched_controls), len(treated_set))
+
+        # Assert it works equally if no covariates are given (automatically assigned)
+        matched_controls_no_covariates_specified = neighbors_matching.match_controls(
+            medrecord=self.medrecord,
+            control_set=control_set,
+            treated_set=treated_set,
+            patients_group="patients",
+        )
+
+        self.assertTrue(matched_controls_no_covariates_specified.issubset(control_set))
+        self.assertEqual(
+            len(matched_controls_no_covariates_specified), len(treated_set)
+        )
+
+    def test_check_nodes(self):
+        neighbors_matching = NeighborsMatching(number_of_neighbors=1)
+
+        control_set: Set[NodeIndex] = {"P1", "P3", "P5", "P7", "P9"}
+        treated_set: Set[NodeIndex] = {"P2", "P4", "P6", "P8"}
+
+        # Test valid case
+        valid_control_set = neighbors_matching._check_nodes(
+            medrecord=self.medrecord,
+            treated_set=treated_set,
+            control_set=control_set,
+            essential_covariates=["age", "gender"],
+        )
+        self.assertEqual(valid_control_set, control_set)
+
+    def test_invalid_check_nodes(self):
+        neighbors_matching = NeighborsMatching(number_of_neighbors=1)
+
+        control_set: Set[NodeIndex] = {"P1", "P3", "P5", "P7", "P9"}
+        treated_set: Set[NodeIndex] = {"P2", "P4", "P6"}
+
+        # Test insufficient control subjects
+        with self.assertRaises(ValueError) as context:
+            neighbors_matching._check_nodes(
+                medrecord=self.medrecord,
+                treated_set=treated_set,
+                control_set={"P1"},
+                essential_covariates=["age", "gender"],
+            )
+        self.assertEqual(
+            str(context.exception),
+            "Not enough control subjects to match the treated subjects",
+        )
+
+        with self.assertRaises(ValueError) as context:
+            neighbors_matching = NeighborsMatching(number_of_neighbors=2)
+            neighbors_matching._check_nodes(
+                medrecord=self.medrecord,
+                treated_set=treated_set,
+                control_set=control_set,
+                essential_covariates=["age", "gender"],
+            )
+        self.assertEqual(
+            str(context.exception),
+            "Not enough control subjects to match the treated subjects",
+        )
+
+        # Test missing essential covariates in treated set
+        with self.assertRaises(ValueError) as context:
+            neighbors_matching._check_nodes(
+                medrecord=self.medrecord,
+                treated_set={"P2", "P10"},
+                control_set=control_set,
+                essential_covariates=["age", "gender"],
+            )
+        self.assertEqual(
+            str(context.exception),
+            "Some treated nodes do not have all the essential covariates",
```

Comment:
Is this correct? Shouldn't it be an Error because 'P10' doesn't exist instead of it not having the covariate

Reply:
Changed to include the P10 inside the `MedRecord` without covariates